src/renderer.ts

```diff
@@ -575,31 +591,32 @@ class Renderer extends EventEmitter<RendererEvents> {
     this.cursor.style.backgroundColor = `${this.options.cursorColor || this.options.progressColor}`
     this.cursor.style.width = `${this.options.cursorWidth}px`
 
-    // Render the waveform
-    if (this.options.splitChannels) {
-      let counter = 0
-      const done = () => {
-        counter++
-        if (counter === audioData.numberOfChannels) {
-          this.emit('rendered')
-        }
-      }
+    this.audioData = audioData
+
+    this.emit('render')
 
-      // Render a waveform for each channel
-      for (let i = 0; i < audioData.numberOfChannels; i++) {
-        const options = { ...this.options, ...this.options.splitChannels[i] }
-        this.renderChannel([audioData.getChannelData(i)], options, width, done)
+    // Render the waveform
+    try {
+      if (this.options.splitChannels) {
+        // Render a waveform for each channel
+        await Promise.allSettled(
+          Array.from({ length: audioData.numberOfChannels }).map((_, i) => {
+            const options = { ...this.options, ...this.options.splitChannels?.[i] }
+            return this.renderChannel([audioData.getChannelData(i)], options, width)
+          }),
+        )
```

Comment:
`allSettled` will prevent the catch from being invoked in subsequent renders.

This seems like more render calls would be happing than one might expect.

I think `Promise.all` might be the better option here. But I may be miss understanding something here.

Reply:
True, thank you, we want to reject the entire promise when one of them fail.